packages/mysql/lib/mysql_p.d.ts

```diff
@@ -10,7 +12,7 @@ declare namespace captureMySQL {
     (options: string, values: any, callback?: MySQL.queryCallback, segment?: AWSXRay.SegmentLike): MySQL.Query;
   }
 
-  type PatchedConnection<T = MySQL.Connection> = {
+  type PatchedConnection<T = MySQL.Connection | MySQL2.Connection> = {
```

Comment:
We currently cannot use `PatchedConnection` for type `MySQL2.Connection`.
- PatchedQueryFunction currently has support for signatures:
```
    (query: MySQL.Query, segment?: AWSXRay.SegmentLike): MySQL.Query;
    (options: string | MySQL.QueryOptions, callback?: MySQL.queryCallback, segment?: AWSXRay.SegmentLike): MySQL.Query;
    (options: string, values: any, callback?: MySQL.queryCallback, segment?: AWSXRay.SegmentLike): MySQL.Query;
```

For mysql2 however, the query function has different signatures:
- See: https://github.com/sidorares/node-mysql2/blob/master/typings/mysql/lib/protocol/sequences/QueryableBase.d.ts#L12-L39

Reply:
I suggest creating new `PatchedConnection` and  `PatchedQueryFunction` types for `mysql2`.

Reply:
Addressed: [854ab49](https://github.com/aws/aws-xray-sdk-node/pull/676/commits/854ab49f262780527a6fed620e3a7148eeb8528b)